pkg/provisioners/gcppubsub/controller/channel/names.go

```diff
@@ -14,30 +14,25 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package util
+package channel
 
 import (
+	"github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
 	"github.com/knative/eventing/pkg/provisioners/utils"
 
 	eventduck "github.com/knative/eventing/pkg/apis/duck/v1alpha1"
 )
 
-// Underscore is used as a separator between the namespace and name so the concatenated version
-// can't accidentally overlap.
+// Underscore is used as a separator so the concatenated version can't accidentally overlap.
 
-// GenerateTopicName generates the GCP PubSub Topic name given the Knative Channel's namespace and
-// name.
-// Note that it must be stable. The controller and dispatcher rely on this being generated
-// identically.
-func GenerateTopicName(channelNamespace, channelName string) string {
-	return utils.TopicName("_", channelNamespace, channelName)
+// generateTopicName generates the GCP PubSub Topic name given the Knative Channel.
+func generateTopicName(c *v1alpha1.Channel) string {
+	return utils.TopicName("_", c.Name, string(c.UID))
```

Comment:
perhaps that needs a rename? or the arguments? I'd still think that a _common_ util for this might be handy, instead of copying code all over the place

Reply:
I removed namespace because I am starting to worry about name length issues. I can't find a documented upper bound on the length of a topic's name. I would be surprised if a Topic's name can be longer than 255 characters, and might even be something more like 63.

'knative-eventing-channel' + namespace + name + UID can get very long. I think K8s names alone tend allow 253 characters. 

Sooner or later we will need to deal with names that are too long. 'kn-{UID}' is almost certainly safe, but virtually useless for a human.

As for reusing `TopicName`, I probably shouldn't use the existing one. Should I make a new `utils.TopicNameWithUID`? That can be common for systems that require the uniqueness of the UID.

Reply:
Made `utis.TopicNameWithUID`.